third_party/txt/src/txt/paragraph_txt.cc

```diff
@@ -429,6 +433,9 @@ bool ParagraphTxt::ComputeBidiRuns(std::vector<BidiRun>* result) {
       if (u_hasBinaryProperty(last_char, UCHAR_WHITE_SPACE)) {
         has_trailing_whitespace = true;
         bidi_run_count--;
+        if (bidi_run_start == 0) {
```

Comment:
IIUC, the goal here is merging the last visual run into the second-to-last run if the last run consists of a single whitespace character.

Would it work to check whether the last run's start index follows the second-to-last run's end index here?  If that is the case, then it's safe to reduce the bidi_run_count and increase the final run's length.

Reply:
Yeah, that should be a more robust check. Implemented.